pkg/kubectl/cmd/set/set_selector.go

```diff
@@ -213,6 +213,9 @@ func updateSelectorForObject(obj runtime.Object, selector metav1.LabelSelector)
 
 // getResourcesAndSelector retrieves resources and the selector expression from the given args (assuming selectors the last arg)
 func getResourcesAndSelector(args []string) (resources []string, selector *metav1.LabelSelector, err error) {
+	if len(args) == 0 {
```

Comment:
I think we need this.  IIUC, this is run by "Complete" before the "Validate" fn to set the resource and selector on "o".